PKHeX/Legality/Core.cs

```diff
@@ -927,7 +927,7 @@ internal static EncounterTrade getValidIngameTrade(PKM pkm, GameVersion gameSour
 
             if (!pkm.WasIngameTrade)
             {
-                if (pkm.HasOriginalMetLocation)
+                if (!pkm.HasOriginalMetLocation)
```

Comment:
I'm thing this should be reverted
If a pokemon has original met location and WasIngameTrade then is impossible for the pokemon to be a be from a valid in game trade, then !pkm.WasIngameTrade, pkm.HasOriginalMetLocation return null
If the pokemon is a gen1/2/3/4 transfer WasIngameTrade will never be true because the variable is based on MetLocation whose data was lost during the transfer, but that does not mean the pokemon could not be originated from a ingame trade